voc/python/ast.py

```diff
@@ -694,7 +712,7 @@ def visit_Try(self, node):
         # Finally content is duplicated at the end of the body
         # if it is defined.
         if node.finalbody:
-            for child in node.finalbody:
+            for child in copy.deepcopy(node.finalbody):
```

Comment:
Why the deep copy?

Reply:
Currently `parse_yield` from previous PR is modifying yield expression nodes in place, if I did not copy the list, subsequent visit to final body's childs (that contains yield expression nodes) will visit the transformed nodes and generate different opcodes.

I'm trying to refactor either `parse_yield` or come up with a way to copy generated opcodes from first visit since using `deepcopy` might not be efficient and optimal when the code is large.

Reply:
I don't know if that's a likely outcome - the body of a yield has to be an expression, not a statement block, so there's a practical limit to how complex a yield body can be.

Reply:
@freakboy3742 I've refactored the code in latest commit so that I'm no longer modify/transform the nodes. However, deepcopy is still needed in the following lines:
 ```        
node_to_resolve = copy.deepcopy(node)  # deepcopy is required for multiple visits of finalbody from visitTry
self.context.yield_points.append(node_to_resolve)
self.context.next_resolve_list.append((node_to_resolve, OpcodePosition.YIELD))
```
in order to resolve the next opcode correctly for duplicated final body as the code snippet below in `blocks.py` changed node's attribute in place:
```
# Resolve any references to the "next" opcode.
for (obj, attr) in self.next_resolve_list:
    # print("        resolve %s reference on %s %s with %s %s" % (
    #     attr, obj, id(obj), opcode, id(opcode))
    # )
    setattr(obj, attr.value, opcode)
```

If everything is fine then it is ready for merge 😃